qa/L0_model_update/test.sh

```diff
@@ -53,33 +53,42 @@ function setup_models() {
         mv models/model_init_del/model.py models/model_init_del/1
 }
 
-SERVER_ARGS="--model-repository=models --model-control-mode=explicit --rate-limit=execution_count --log-verbose=2"
-SERVER_LOG="./server.log"
-setup_models
-run_server
-if [ "$SERVER_PID" == "0" ]; then
-    echo -e "\n***\n*** Failed to start $SERVER\n***"
-    cat $SERVER_LOG
-    exit 1
-fi
+# Test model instance update with and without rate limiting enabled
+for RATE_LIMIT_MODE in "off" "execution_count"; do
 
-RET=0
+    export RATE_LIMIT_MODE=$RATE_LIMIT_MODE
+    TEST_LOG="instance_update_test.rate_limit_$RATE_LIMIT_MODE.log"
+    SERVER_LOG="./instance_update_test.rate_limit_$RATE_LIMIT_MODE.server.log"
 
-set +e
-python instance_update_test.py > instance_update_test.log 2>&1
-if [ $? -ne 0 ]; then
-    cat instance_update_test.log
-    RET=1
-fi
-set -e
+    setup_models
+    SERVER_ARGS="--model-repository=models --model-control-mode=explicit --rate-limit=$RATE_LIMIT_MODE --log-verbose=2"
+    run_server
+    if [ "$SERVER_PID" == "0" ]; then
+        echo -e "\n***\n*** Failed to start $SERVER\n***"
+        cat $SERVER_LOG
+        exit 1
+    fi
 
-kill $SERVER_PID
-wait $SERVER_PID
+    RET=0
 
-if [ $RET -eq 0 ]; then
-    echo -e "\n***\n*** Test Passed\n***"
-else
-    echo -e "\n***\n*** Test FAILED\n***"
-fi
+    set +e
+    python instance_update_test.py > $TEST_LOG 2>&1
+    if [ $? -ne 0 ]; then
+        cat $TEST_LOG.log
+        echo -e "\n***\n*** Test FAILED\n***"
+        RET=1
+    fi
+    set -e
+
+    kill $SERVER_PID
+    wait $SERVER_PID
+
+    if [ $RET -ne 0 ]; then
+        exit $RET
```

Comment:
Why are we exiting on the first failure? 
Shouldn't we still execute "execution_count" mode even if the "off" mode failed?

Reply:
> Why are we exiting on the first failure?

This is to make finding the location of the failure easier on the CI, because it is not covered by other tests.

 > Shouldn't we still execute "execution_count" mode even if the "off" mode failed?
 
 Yes, we can continue execute if "off" mode failed.

Reply:
Updated: [Continue test if off mode failed](https://github.com/triton-inference-server/server/pull/5724/commits/e63f7d40f01650a44f522caa0865820e040b8c1c)